lib/std/process.zig

```diff
@@ -775,6 +775,24 @@ pub fn getSelfExeSharedLibPaths(allocator: *Allocator) error{OutOfMemory}![][:0]
             }
             return paths.toOwnedSlice();
         },
+        // revisit if Haiku implements dl_iterat_phdr (https://dev.haiku-os.org/ticket/15743)
```

Comment:
You could also submit a patch to Haiku implementing it ;)

Otherwise, yes, `get_next_image_info` is the way to go.

Reply:
thanks for the invitation :) and might consider it in the future if someone else doesn't beat me to it. for the nearer term, i prefer to keep my focus on having something that can get this in a better compiling state on Haiku